cmake/cc_rules.cmake

```diff
@@ -329,6 +354,43 @@ function(objc_framework target)
   endif()
 endfunction()
 
+function(objc_test target)
+  if(APPLE)
+    set(flag EXCLUDE_FROM_ALL)
+    set(single HOST VERSION)
+    set(multi DEPENDS DEFINES HEADERS INCLUDES SOURCES)
+    cmake_parse_arguments(ot "${flag}" "${single}" "${multi}" ${ARGN})
+
+    xctest_add_bundle(
+      ${target}
+      ${ot_HOST}
+      ${ot_SOURCES}
+    )
+
+    add_objc_flags(
+      ${target}
+      ${ot_SOURCES}
+    )
+
+    target_link_libraries(
+      ${target}
+      PRIVATE
+        ${ot_DEPENDS}
+    )
+
+    xctest_add_test(
+      ${target}
+      ${target}
+    )
+
+    if(WITH_ASAN)
+      set_property(TEST ${target} APPEND PROPERTY
+        ENVIRONMENT
+        DYLD_INSERT_LIBRARIES=/usr/local/opt/llvm/lib/clang/8.0.0/lib/darwin/libclang_rt.asan_osx_dynamic.dylib)
```

Comment:
I was able to fix the asan issue locally with this change. However, I could not figure out a way to set this ENV variable reliably both on my local machine and on travis. 

Ideally the compiler could tell us where this `libclang_rt.asan_osx` is, or we could infer from the compiler being used.